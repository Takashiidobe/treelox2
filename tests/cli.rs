use std::process::Output;

use assert_cmd::cargo::CommandCargoExt;
use insta::{assert_yaml_snapshot, glob};
use insta_cmd::Command;
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize)]
struct TestOutput {
    status: i32,
    stdout: Vec<String>,
    stderr: Vec<String>,
}

#[test]
fn reference_files() {
    glob!("../test-files", "**/*.lox", |path| {
        let mut cmd = Command::cargo_bin(env!("CARGO_PKG_NAME")).unwrap();

        let Output {
            status,
            stdout,
            stderr,
        } = cmd.arg(path).output().unwrap();

        let test_output = TestOutput {
            status: status.code().unwrap(),
            stdout: String::from_utf8_lossy(&stdout)
                .to_string()
                .lines()
                .map(|x| x.to_owned())
                .collect(),
            stderr: String::from_utf8_lossy(&stderr)
                .to_string()
                .lines()
                .map(|x| x.to_owned())
                .collect(),
        };

        assert_yaml_snapshot!(test_output);
    });
}
