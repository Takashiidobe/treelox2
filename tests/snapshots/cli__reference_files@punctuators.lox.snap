---
source: tests/cli.rs
expression: test_output
input_file: test-files/scanning/punctuators.lox
---
status: 0
stdout:
  - LeftParen ( on line 1
  - RightParen ) on line 1
  - "LeftBrace { on line 1"
  - "RightBrace } on line 1"
  - Semicolon ; on line 1
  - "Comma , on line 1"
  - Plus + on line 1
  - Minus - on line 1
  - Star * on line 1
  - BangEqual != on line 1
  - EqualEqual == on line 1
  - LessEqual <= on line 1
  - GreaterEqual >= on line 1
  - BangEqual != on line 1
  - Less < on line 1
  - Greater > on line 1
  - Slash / on line 1
  - Dot . on line 1
  - Eof EOF on line 22
stderr: []
